locals.tf

```diff
@@ -0,0 +1,27 @@
+# ------------------------------------------------------------------------------
+# Retrieve the effective Account ID, User ID, and ARN in which Terraform is
+# authorized.  This is used to calculate the session names for assumed roles.
+# ------------------------------------------------------------------------------
+data "aws_caller_identity" "default" {}
+
+# ------------------------------------------------------------------------------
+# Evaluate expressions for use throughout this configuration.
+# ------------------------------------------------------------------------------
+locals {
+  # Extract the user name of the current caller for use
+  # as assume role session names.
+  caller_user_name = split("/", data.aws_caller_identity.default.arn)[1]
+
+  # Assumption of the following roles is required to successfully manage IAM
+  # users and groups.
+  required_roles = [
```

Comment:
The reason this is like this is because the repos that we use to manage users and groups rely on assumption of these roles.  For reference, here are the providers in the repos that use those roles:

- https://github.com/cisagov/cool-users-non-admin/blob/develop/providers.tf
- https://github.com/cisagov/cool-admin-provisioner-iam/blob/develop/providers.tf
- https://github.com/cisagov/cool-assessment-provisioner-iam/blob/develop/providers.tf
- https://github.com/cisagov/cool-auditor-iam/blob/develop/providers.tf
- https://github.com/cisagov/cool-ses-send-email-iam/blob/develop/providers.tf

If we were to modify those repos to use more specific (i.e. less powerful) roles, then we could change the roles here.

For now, how about I add a comment that explains all of this a bit more and provides links to those providers files?

Reply:
Alright. My main concern is that [cisagov/provisionaccount-role-tf-module](https://github.com/cisagov/provisionaccount-role-tf-module) includes the `IAMFullAccess` managed policy which, per the [JSON document](https://docs.aws.amazon.com/aws-managed-policy/latest/reference/IAMFullAccess.html#IAMFullAccess-json), includes the `iam:*` permission.

Reply:
Let me know how https://github.com/cisagov/cool-iam-user-group-manager-iam/pull/4/commits/c3fabef279d4ab336ad164968b0604f3ffb1e4fa works for you.

Reply:
@mcdonnnj Per our chat earlier, please see https://github.com/cisagov/cool-system-internal/issues/137 and https://github.com/cisagov/cool-iam-user-group-manager-iam/pull/4/commits/5d44fbb89adde7ec74283436631221120c937d7c.  Let me know if that is sufficient to resolve this conversation.

Reply:
Yes that works for me. Thanks for creating that issue and updating the comment 🏆